recipe/meta.yaml

```diff
@@ -22,7 +22,9 @@ build:
   # This can be removed in a later CUDA Python that supports Cython 3.x.
   # xref: https://github.com/conda-forge/cuda-python-feedstock/issues/57
   skip: true  # [py == 312]
-  skip: true  # [cuda_compiler_version != "12.1"]
+  {% if not (environ.get("cuda_compiler_version")|string()).startswith(major_version|string()) %}
+  skip: true
+  {% endif %}
```

Comment:
Have altered the `skip` logic here to be independent of a specific CUDA compiler version. Instead it focuses on matching the CTK major version with the package version. As we are only pinning the CUDA compiler major version in the migrator, this skip will select out the one relevant builds (CUDA 12) and ignore all other ones

Pinning the CTK minor version is handled in the requirements (via `cuda-version`) instead of the `skip`, which hopefully will make it easier to follow and simpler to maintain

This way the `skip` details go into the background and the recipe can focus on the one important thing. Namely building `cuda-python` with a matching CTK version (handled via `cuda-version`)